src/libs/SearchParser/searchParser.peggy

```diff
@@ -5,147 +5,162 @@
     return {
       ...defaultValues,
-      filters
-    };
-  }
-
-  function applyPolicyID(filtersWithDefaults) {
-    return {
-      ...filtersWithDefaults,
-      policyID: filtersWithDefaults.policyID
+      filters,
     };
   }
 
   function updateDefaultValues(field, value) {
     defaultValues[field] = value;
   }
-
-  function isDefaultField(field) {
-    return defaultValues.hasOwnProperty(field);
-  }
-
-  function isPolicyID(field) {
-    return field === 'policyID';
-  }
 }
 
-query
-  = _ filters:filterList? _ {
-    const withDefaults = applyDefaults(filters);
-    if (defaultValues.policyID) {
-        return applyPolicyID(withDefaults);
-    }
-
-    return withDefaults;
-  }
+query = _ filters:filterList? _ { return applyDefaults(filters); }
 
 filterList
   = head:filter tail:(logicalAnd filter)* {
-      const allFilters = [head, ...tail.map(([_, filter]) => filter)].filter(filter => filter !== null);
+      const allFilters = [head, ...tail.map(([_, filter]) => filter)]
+        .filter((filter) => filter !== null)
+        .filter(Boolean);
       if (!allFilters.length) {
-      	return null;
+        return null;
+      }
+
+      const keywords = allFilters.filter(
+        (filter) =>
+          filter.left === "keyword" || filter.right?.left === "keyword"
+      );
+      const nonKeywords = allFilters.filter(
+        (filter) =>
+          filter.left !== "keyword" && filter.right?.left !== "keyword"
+      );
+
+      if (!nonKeywords.length) {
+        return keywords.reduce((result, filter) =>
+          buildFilter("or", result, filter)
+        );
       }
-     const keywords = allFilters.filter((filter) => filter.left === "keyword" || filter.right?.left === "keyword")
-     const nonKeywords = allFilters.filter((filter) => filter.left !== "keyword" && filter.right?.left !== "keyword")
-     if(!nonKeywords.length){
-     	return keywords.reduce((result, filter) => buildFilter("or", result, filter))
-     }
-     if(!keywords.length){
-     	return nonKeywords.reduce((result, filter) => buildFilter("and", result, filter))
-     }
-
-     return buildFilter("and", keywords.reduce((result, filter) => buildFilter("or", result, filter)), nonKeywords.reduce((result, filter) => buildFilter("and", result, filter)))
-
-      return allFilters.reduce((result, filter) => buildFilter("and", result, filter));
+      if (!keywords.length) {
+        return nonKeywords.reduce((result, filter) =>
+          buildFilter("and", result, filter)
+        );
+      }
+
+      return buildFilter(
+        "and",
+        keywords.reduce((result, filter) => buildFilter("or", result, filter)),
+        nonKeywords.reduce((result, filter) =>
+          buildFilter("and", result, filter)
+        )
+      );
     }
 
 filter
-  = _ field:key? _ op:operator? _ value:identifier {
-      if (isDefaultField(field)) {
-        updateDefaultValues(field, value.trim());
-        return null;
-      }
+  = @(
+      policyIDFilter
```

Comment:
Why do we need a separate policyIDFilter? Can't we add the policyID key to the `key` list and use the defaultFilter rule?

Reply:
@289Adam289 will provide more context I'm sure, but we don't want to group policyID together. There is no default value for `policyID` and specifically we don't want to land in a case where we get object looking like this:
```
{
   type: Expense,
   [..]
   policyID: undefined
}
```

And I've seen this case on some other versions. If you have key with value undefined then two things happen:
 - we will iterate over it
 - we might accidentally stringify it to get `policyID:undefined...`

Reply:
I think you are right @luacmartins. I missed that `policyId` is running the same code as default filters. Adding `policyID` to `defaultKeys` and not adding it to `defaultValues` should work.